### ISSUE TITLE
tdweb api status

### ISSUE THREAD

**Issue by 1343322562** [tdlib/td#3099]
TDweb used 'send' API, the promise status has always been in 'pending', I hope to return a status of fulfilled or reject, but nothing.
I used VPN to connect to the Telegram on WEB, but i can't used the TDweb
![e468f888613b074a32376f0065c769a](https://github.com/user-attachments/assets/7f82ca56-1a29-4fcf-bf36-b22bbc0d4535)

**Comment by levlam** [comment]
You can enable TDLib logs using `options.logVerbosityLevel=4` to see, what exactly happens with the request. Likely, it can't reach Telegram servers.

**Comment by 1343322562** [comment]
> You can enable TDLib logs using `options.logVerbosityLevel=4` to see, what exactly happens with the request. Likely, it can't reach Telegram servers.

I think so, but i don't know how to do it, Cause i have connected VPN in Hong Kong, maybe i need to switch VPN address. appreciate you

**Comment by levlam** [comment]
What app do you use?

**Comment by Scutua** [comment]
Yes +66842760493

**Comment by levlam** [comment]
`tdweb` has no user interface and you shown a complete log in interface on the screenshot.

**Comment by 1343322562** [comment]
yes. It comes from 'https://github.com/evgeny-nadymov/telegram-react.git', an example of tdweb

**Comment by levlam** [comment]
Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.

**Comment by 1343322562** [comment]
> Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.
The highest version of `tdweb` is 1.8.0.
`tdlib` should be integrated into `tdweb`.
How should i up it, or `tdweb` is unvailable

**Comment by 1343322562** [comment]
> Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.

Then, you can add the parameter verbosity=4 to the URL, but telegram-react uses a very old TDLib version and will not be able to log in anyway now.
The highest version of `tdweb` is 1.8.0.
`tdlib` should be integrated into `tdweb`.
How should i up it, or `tdweb` is unvailable

**Comment by 1343322562** [comment]
> Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.

Then, you can add the parameter verbosity=4 to the URL, but telegram-react uses a very old TDLib version and will not be able to log in anyway now.
The highest version of `tdweb` is 1.8.0.
`tdlib` should be integrated into `tdweb`.
How should i up it, or `tdweb` is unvailable

**Comment by 1343322562** [comment]
> Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.


The highest version of `tdweb` is 1.8.0.
`tdlib` should be integrated into `tdweb`.
How should i up it, or `tdweb` is unvailable

**Comment by 1343322562** [comment]
> Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.

I saw the last submission was 3 years ago
![Uploading 9f011dc9fdf80b90653826fc05b8906.png…]()

**Comment by 1343322562** [comment]
> Then, you can add the parameter `verbosity=4` to the URL, but `telegram-react` uses a very old TDLib version and will not be able to log in anyway now.

I saw that `tdweb` was last submitted 3 years ago. it's unvailable ?

**Comment by Scutua** [comment]
Can't it said something went wrong , possible at me what app make easy to
talk

**Comment by levlam** [comment]
You can build TDLib for web yourself following https://github.com/tdlib/td/tree/master/example/web, but then you will need to update the telegram-react code to support all changes in TDLib API, and there were a lot of API changes since then.

**Comment by 1343322562** [comment]
> You can build TDLib for web yourself following https://github.com/tdlib/td/tree/master/example/web, but then you will need to update the telegram-react code to support all changes in TDLib API, and there were a lot of API changes since then.

I got it.
I'll try to manually build it, thanks

**Comment by lairwaves-dev** [comment]
> > You can build TDLib for web yourself following https://github.com/tdlib/td/tree/master/example/web, but then you will need to update the telegram-react code to support all changes in TDLib API, and there were a lot of API changes since then.
> 
> I got it. I'll try to manually build it, thanks

I am encountering an "UPDATE_APP_TO_LOGIN" error when trying to run the Telegram web client. I'm following the instructions to build TDLib for web myself, but I'm not sure if this error is related to the TDLib version or some other compatibility issue. Do you have any experience with this specific error or suggestions on how to troubleshoot it? I appreciate any guidance you can provide.

**Comment by levlam** [comment]
@lairwaves-dev You use an old TDLib version and not the latest version that was built manually. Check response to getOption("version") to find the used version.

**Comment by lairwaves-dev** [comment]
> [@lairwaves-dev](https://github.com/lairwaves-dev) You use an old TDLib version and not the latest version that was built manually. Check response to getOption("version") to find the used version.

Appreciate your quick response! I really appreciate your help. I'll check the getOption("version") response to verify the TDLib version and try building it again.

**Comment by 0xdsaini** [comment]
even version check is showing Promise(<pending>). Apparently all API requests are going that way.

**Comment by 0xdsaini** [comment]
Update: It worked!

The latest version hosted on npm is 1.8.0 which doesn't work and produces "UPDATE_APP_TO_LOGIN" error. You need to build the latest tdweb from their github repository https://github.com/tdlib/td.git
